core/src/main/java/org/stellar/anchor/sep24/Sep24Service.java

```diff
@@ -252,10 +248,13 @@ public InteractiveTransactionResponse withdraw(
     }
 
     String quoteId = withdrawRequest.get("quote_id");
+    AssetInfo buyAsset = assetService.getAssetByName(withdrawRequest.get("destination_asset"));
     if (quoteId != null) {
-      AssetInfo buyAsset = assetService.getAssetByName(withdrawRequest.get("destination_asset"));
-      this.validatedAndPopulateQuote(
+      validateAndPopulateQuote(
           quoteId, asset, buyAsset, strAmount, builder, WITHDRAWAL.toString(), txnId);
+    } else {
+      builder.amountExpected(strAmount);
+      if (buyAsset != null) builder.amountOutAsset(buyAsset.getSep38AssetName());
```

Comment:
Should it not be `amountInAsset`?

Reply:
In `withdraw`, the buy should be `amountOutAsset`?

ex: sell USDC and buy USD. So, the `amountOutAsset` should be USD.

Reply:
I didn't see it was withdraw, you are right